### PR TITLE
[WIP] Fetched local mercurial repo's commit hash

### LINKED ISSUE #1506: Instead of applying patch in git, fetch it from the local Mercurial repository
Using git-cinnabar, we can replace https://github.com/mozilla/bugbug/blob/e60a51b6b744ae1c1bbd65cbc28fc7f796f851e0/scripts/commit_classifier.py#L358-L380 and just fetch the commit from the local Mercurial repository where we already applied the patch.

Then, we can stop installing `patch` in our Docker image: https://github.com/mozilla/bugbug/blob/e60a51b6b744ae1c1bbd65cbc28fc7f796f851e0/infra/dockerfile.commit_retrieval#L10.

### PR DESCRIPTION
**Pull request by Dcoder99**
closes #1506 
I have currently only fetched the local mercurial repo's commit hash where the patches are applied, yet to apply that commit to the git repo and replace lines 358-380.
I hope I'm in the right direction :)

### CODE REVIEW

File: scripts/commit_classifier.py [hunk 1]
```diff
@@ -355,6 +355,31 @@ def load_user(phid):
...
+                
+                # Finding 1st occurrence of ',' 
+                substr = ","
+                occurrence = 1
+                inilist = [i for i in range(0, len(ini_str)) 
+                            if ini_str[i:].startswith(substr)] 
+                end = inilist[occurrence - 1]
+
+                # hash of the local commit
+                local_commit_hash = res[start:end]
```
Comment (marco-c) [inline scripts/commit_classifier.py:381]: You should be able to replace all this by using hg.log instead, taking the last applied revisions. You can use https://github.com/mozilla/bugbug/blob/c8ad32b2d6480f65132208448b21cf3eda3a46c8/bugbug/repository.py#L687.

So, first apply all patches to the local mercurial repository, then after the end of the `for patch in needed_stack:` loop, you can get the applied revisions with `get_revs(hg, f"-{len(stack)}")`.

Comment (Dcoder99) [reply scripts/commit_classifier.py:381]: Thanks, will do.
But the first comment of #1506 issue says **_"just fetch the commit from the local Mercurial repository where we already applied the patch."_**
So I am assuming that the patches are already applied to the local hg repo. So I will just need to use get_revs to get the hash of latest revision. Is that correct?

Comment (marco-c) [reply scripts/commit_classifier.py:381]: The code which is applying the patch is the `for patch in needed_stack:` loop I mentioned above :)

File: scripts/commit_classifier.py [hunk 2]
```diff
@@ -315,10 +315,12 @@ def load_user(phid):
                 revision["fields"]["title"], revision["fields"]["summary"]
             )
 
+            latest_rev = repository.get_revs(hg, f"-{len(stack)}")
```
Comment (marco-c) [inline scripts/commit_classifier.py:318]: We haven't applied the patches yet here, so this will return wrong revisions.

Comment (marco-c) [reply scripts/commit_classifier.py:318]: The code should moved after the end of the for loop.

Comment (Dcoder99) [reply scripts/commit_classifier.py]: Done!

File: scripts/commit_classifier.py [hunk 3]
```diff
@@ -377,6 +373,8 @@ def load_user(phid):
                     check=True,
                     cwd=self.git_repo_dir,
                 )
+        latest_rev = repository.get_revs(hg, f"-{len(stack)}")
```
Comment (marco-c) [inline scripts/commit_classifier.py:376]: You're setting this variable here, but trying to use it earlier :smile: 
Maybe this is not an easy enough task to start with, I'd suggest working on some personal projects first to get some initial experience with programming.

**Pull request closed without merging**